packages/ag-charts-community/src/util/stateMachine.ts

```diff
@@ -115,6 +115,13 @@ export class StateMachine<State extends string, Event extends string> {
         }
     }
 
+    transitionAsync(event: Event, data?: any) {
+        // TODO: consider merging this with transition
+        setTimeout(() => {
+            this.transition(event, data);
+        }, 0);
+    }
```

Comment:
@alantreadway I discussed this with Mana.

The use-case of this was to ensure the transition to the new state happens after the current tick, since both the original tick and the transition were manipulating the `postUpdateFns` array in `annotations.ts`.

It is perhaps prefereable that all transitions should be asynchronous (and queued), but that's wayyy too risky to do now.

Reply:
Agree, we should probably do a sweep post release for `setTimeout()`, `requestAnimationFrame()` and try to work out if there is a common pattern/approach we should adopt.